src/transformers/generation_utils.py

```diff
@@ -629,9 +633,15 @@ def _get_logits_processor(
         return processors
 
     def _get_stopping_criteria(
-        self, max_length: Optional[int], max_time: Optional[float], max_new_tokens: Optional[int], start_length: int
+        self,
+        max_length: Optional[int],
+        max_time: Optional[float],
+        max_new_tokens: Optional[int],
+        start_length: int,
+        stopping_criteria: Optional[StoppingCriteriaList],
     ) -> StoppingCriteriaList:
-        stopping_criteria = StoppingCriteriaList()
+        if stopping_criteria is None:
+            stopping_criteria = StoppingCriteriaList()
         if max_length is not None:
```

Comment:
max_length always defaults to `20` -> so if someone passes a `stopping_criteria` list then there are two stopping criteria no?

Reply:
This is not good no? E.g. if the stopping criteria is 30 in the list, the generation will still stop at 20. So IMO if someone passes a `stopping_criteria` list we should check for each item if the class already exsits in the list (if it's not the case, only then we'll add it). This means that the priority is as follows:

1st priority: `stopping_criteria`
2nd priority: directly passing `max_length`
3rd priority: using `max_length` of config
4th priority: using default `max_length` 

=> think the same should hold true for `logits_processor`.

Think we should not merge as it is right now